src/dbnode/digest/writer.go

```diff
@@ -64,13 +64,18 @@ func (w *fdWithDigestWriter) Write(b []byte) (int, error) {
 	return written, nil
 }
 
-// Close flushes what's remaining in the buffered writer and closes
+// Close flushes what's remaining in the buffered writer and syncs and closes
 // the underlying file.
 func (w *fdWithDigestWriter) Close() error {
 	if err := w.writer.Flush(); err != nil {
 		return err
 	}
-	return w.FdWithDigest.Close()
+	syncErr := w.FdWithDigest.Fd().Sync()
```

Comment:
Do we have any data to see how this affects performance?

Reply:
This is a very good question. I took some real-world timings:
```
closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-info.db
flush took 13.445µs
sync took 2.545475ms
close took 83ns

closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-index.db
flush took 56.619µs
sync took 13.255645ms
close took 87ns

closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-summaries.db
flush took 73.178µs
sync took 2.316232ms
close took 82ns

closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-bloomfilter.db
flush took 34.051µs
sync took 1.641726ms
close took 79ns

closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-data.db
flush took 42.687µs
sync took 9.755168ms
close took 88ns

closing /var/lib/m3db/snapshots/default/250/fileset-1603432800000000000-6-digest.db
flush took 22.099µs
sync took 2.024822ms
close took 80ns
```
If these numbers are concerning, we could parallelize persisting of these 6 files that can be handled in any order:
https://github.com/m3db/m3/blob/20850befd7daa47faa2f0bcc10ebdbd042ec0a02/src/dbnode/persist/fs/write.go#L393-L400